disable cephObjectStore for IBMCloud

Gangadhar-ibm opened the pull request:
CephObjectStore is disabled for all cloud platforms
Earlier, we were disabling it for IBMCloud only if COS was used as default backing store
However, with this change, we are disabling it for IBM Cloud platform irrespective of COS being used for backing store

akgunjal commented:
@jarrpa @iamniting : The purpose of this PR is to stop creating the storage class `ocs-storagecluster-ceph-rgw` on IBM Cloud as its not needed. But we do want to retain the support we have today for COS as backing storage. So please let us know if the change in this PR will remove that support as well.

Shirisha-S-Rao-1 commented:
@iamniting 
@petr-balogh Informed us that we don't need the rgw class and related resources for any cloud platform. So, we want to remove it for IBM Cloud

When we made the COS secret changes, we structured it such that, if COS secret was present, we weren't creating the rgw class and only creating it in the absence of the COS secret.
But now, we want to remove rgw for IBM Cloud since it is not needed on cloud platforms

iamniting commented:
> @iamniting @petr-balogh Informed us that we don't need the rgw class and related resources for any cloud platform. So, we want to remove it for IBM Cloud
> 
> When we made the COS secret changes, we structured it such that, if COS secret was present, we weren't creating the rgw class and only creating it in the absence of the COS secret. But now, we want to remove rgw for IBM Cloud since it is not needed on cloud platforms

I see, instead of this I would say why don't we extend this list to include IBM as well. It will automatically skip the SC as it does for others. https://github.com/red-hat-storage/ocs-operator/blob/1904024aaa230f8a868f4bf2cde7879210b41630/controllers/storagecluster/platform_detection.go#L26

Gangadhar-ibm commented on controllers/storagecluster/platform_detection.go:
```diff
@@ -27,7 +27,7 @@ var SkipObjectStorePlatforms = []configv1.PlatformType{
 	configv1.GCPPlatformType,
 	configv1.AzurePlatformType,
-	IBMCloudCosPlatformType,
+	configv1.IBMCloudCosPlatformType,
```
I have changed and pushed changes

iamniting replied:
I mean both of them should be added to the list so that both of them will be ignored.
```
configv1.IBMCloudCosPlatformType
configv1.IBMCloudPlatformType
```

Gangadhar-ibm replied:
I have added both of them to the list

iamniting commented:
> @iamniting IBMCloudCosPlatformType is a subset of IBMCloudPlatformType Therefore, adding IBMCloudPlatformType is enough

if that's a case we can remove the other one.

iamniting commented:
@Gangadhar-ibm Can you pls update the description and commit-msg as well with the details.

Gangadhar-ibm commented:
@iamniting : I have updated the description and commit message. Thanks

iamniting commented:
/retest

iamniting reviewed (approved):
/lgtm

iamniting commented:
@Gangadhar-ibm Can you pls look at the failures and fix them?

for example
```
controllers/storagecluster/platform_detection.go:11:2: imported and not used: "k8s.io/apimachinery/pkg/api/errors"
make: *** [Makefile:66: build-go] Error 2
```

iamniting commented on controllers/storagecluster/platform_detection.go:
```diff
@@ -6,18 +6,12 @@ import (
-	ibmCloudCosSecretName = "ibm-cloud-cos-creds"
-
 	//IBMCloudPlatform with COS Secret
 	IBMCloudCosPlatformType configv1.PlatformType = "IBMCloudCosPlatform"
```
let's remove this constant as well wherever it is used as we are not setting this anymore

Gangadhar-ibm replied:
Done

Gangadhar-ibm commented:
@iamniting: IBMCloudCosPlatformType is using in controllers/storagecluster/cephcluster.go and controllers/storagecluster/cephcluster_test.go

iamniting commented:
> @iamniting: IBMCloudCosPlatformType is using in controllers/storagecluster/cephcluster.go and controllers/storagecluster/cephcluster_test.go

and we should not require those anymore as we are not setting this platform at all

Gangadhar-ibm commented:
> > @iamniting: IBMCloudCosPlatformType is using in controllers/storagecluster/cephcluster.go and controllers/storagecluster/cephcluster_test.go
> 
> and we should not require those anymore as we are not setting this platform at all

HI @iamniting 
I have removed reference of IBMCloudCosPlatformType and updated PR. Please review the PR

iamniting reviewed (approved):
/lgtm

iamniting commented:
/retest-required

iamniting commented:
/test ocs-operator-ci
